File: example-projects/simple-setup/gradle-junit-4-and-5/src/main/java/com/example/ExploreMe.java
package com.example;

public class ExploreMe {
    private final int a;

    public ExploreMe(int a) {
        this.a = a;
    }

	public String exploreMe(int b, String c) {
		if (b == 436) {
			return "Hello";
		}

		if (a >= 20000) {
			if (b >= 2000000) {
				if (b - a < 100000) {
					// Trigger Remote Code Execution Bug
					if (c.startsWith("@")) {
						String className = c.substring(1);
						try {
							Class.forName(className).getDeclaredConstructor().newInstance();
						} catch (ReflectiveOperationException ignored) {
						}
					}
				}
			}
		}
		return c;
	}
}

File: example-projects/simple-setup/gradle-kotlin/src/main/kotlin/com/example/ExploreMe.kt
package com.example

class ExploreMe(private val a: Int) {
    // Function with multiple paths that can be discovered by a fuzzer.
    fun exploreMe(b: Int, c: String) {
        if (a >= 20000) {
            if (b >= 2000000) {
                if (b - a < 100000) {
                    // Trigger Remote Code Execution Bug
                    if (c.startsWith("@")) {
                        val className = c.substring(1)
                        try {
                            Class.forName(className).getDeclaredConstructor().newInstance()
                        } catch (ex: Exception) {
                            when(ex) {
                                is ReflectiveOperationException -> {
                                    // ignore
                                }
                                else -> throw ex
                            }
                        }
                    }
                }
            }
        }
    }
}
